src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -818,6 +823,11 @@ private bool InferNamedPipesInformation()
                         return false;
                     }
 
+                    if(tokensByBackSlash[4].StartsWith(NamedPipeInstanceNameHeader))
```

Comment:
For connection string `\\.\pipe\MSSQL$INSTANCE1\sql\query`, Native SNI gives `MSSQLSvc/GELEE-VM-WIN10.redmond.corp.microsoft.com:instance1` which tells `INSTANCE1` was parsed out during Named Pipe SPN process.
So we do not want the Native SNI's behavior?